tests/test_modules.py

```diff
@@ -196,3 +197,20 @@ def test_import_cache(mock_user_cache_dir, tmp_path):
     assert len(mock_logger.info.call_args_list) == 1
     assert len(list(tmp_path.iterdir())) == n_cached_paths
     mock_iffm.assert_called_once()
+
+@mock.patch.dict(os.environ, {"PYFLYBY_DISABLE_CACHE": "1"})
+@mock.patch("platformdirs.user_cache_dir")
+def test_import_perms(mock_user_cache_dir, tmp_path):
```

Comment:
This covers the case where the directory on sys.path has different permissions. Do we also need to cover the case where _contents_ of the directory have different permissions that the directory itself?